flask_api.py

```diff
@@ -127,31 +147,41 @@ def handle_question():
     server are:
         * storage of the logs of this question-answer-session.
     """
-    init_nimbus_db()
+    try:
 
-    if request.is_json is False:
-        return "request must be JSON", BAD_REQUEST
+        init_nimbus_db()
 
-    request_body = request.get_json()
+        if request.is_json is False:
+            return "request must be JSON", BAD_REQUEST
 
-    question = request_body.get("question", None)
+        request_body = request.get_json()
 
-    if "question" not in request_body:
-        return "request body should include the question", BAD_REQUEST
+        question = request_body.get("question", None)
 
-    try:
-        feedback_saved = db.insert_entity(QuestionLog, {"question": question})
-    except (Exception) as e:
-        print("Could not store question upon user ask: ", str(e))
+        if "question" not in request_body:
+            return "request body should include the question", BAD_REQUEST
 
-    response = {"answer": nimbus.answer_question(question)}
+        try:
+            entity = db.insert_entity(QuestionLog, {"question": question})
+        except (Exception) as e:
+            print("Could not store question upon user ask: ", str(e))
 
-    if "session" in request_body:
-        response["session"] = request_body["session"]
-    else:
-        response["session"] = generate_session_token()
+        try:
+            response = {"answer": nimbus.answer_question(question)}
+        except (Exception) as e:
+            response = {"answer": "Sorry, unable to provide answer to question. Try another question"}
 
-    return jsonify(response), SUCCESS
+        if "session" in request_body:
+            response["session"] = request_body["session"]
+        else:
+            response["session"] = generate_session_token()
+
+        return jsonify(response), SUCCESS
+
+    except Exception as e:
+        log_error(e, question)
+        response = {"answer": "oops, something went wrong... Try another question"}
+        return jsonify(response), SERVER_ERROR # Should this really be SERVER_ERROR? 
```

Comment:
Theoretically, we should determine if it is a SERVER_ERROR or a CLIENT_ERROR.  I'm not sure that's in scope for this - we used to simply 500 and return nothing useful, so this is an improvement.

Reply:
One way I think you could do that is to have two top level error "interfaces" or parent modules/classes and have other errors fall into one of those categories but I agree that the changes necessary to make that happen would fall outside of the scope of this code review